src/App.tsx

```diff
@@ -1,14 +1,62 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isDataLoading, setIsDataLoading] = useState(true);
+  const [isLoadingError, setIsLoadingError] = useState(false);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [todosFilter, setTodosFilter] = useState('all');
+
+  let visibleTodos = todos.filter(todo => {
+    switch (todosFilter) {
+      case 'active':
+        return !todo.completed;
+      case 'completed':
+        return todo.completed;
+      case 'all':
+        return true;
+      default:
+        throw new Error('Wrong todo status');
+    }
+  });
```

Comment:
If you throw an error to the page, it will crash your Application( If you type this field via enum, it will prevent unexpected cases, and you can handle `all` case as default